### ISSUE TITLE
[BUG] `MaskedCategorical`'s `use_cross_entropy` is not compatible with previously working shapes

### ISSUE THREAD

**Issue by LCarmi** [pytorch/rl#3078]
## Describe the bug

The update to `torchrl>=0.9.0` switched `MaskedCategorical`'s default argument from `use_cross_entropy=False` to `use_cross_entropy=True`. This is not compatible with shapes that were previously compatible

## To Reproduce

```python
from functools import partial

import pytest
import torch
from torchrl.modules import MaskedCategorical

def test__MaskedCategorical__fails() -> None:
    """Checks the presence of a shape-related bug when computing log probabilities."""
    batch_size = (2, 3, 4) # typical tensor shape [N, T, A] ie, (batch_size, sequence_length, num_agents)

    masks = torch.tensor([[True, False, True, False, True]], dtype=torch.bool)
    logits = torch.randn((5,)).expand(*batch_size, -1)  # has shape [N, T, A, num_actions] during train
    action = torch.tensor(0, dtype=torch.long).expand(*batch_size)  # has shape [N, T, A] during train

    cat = MaskedCategorical(logits=logits, mask=masks)
    with pytest.raises(RuntimeError, match="size mismatch"):
        cat.log_prob(action)

    fixed_cat = MaskedCategorical(logits=logits, mask=masks, use_cross_entropy=False)
    fixed_cat.log_prob(action)
```

I have tried different shapes for actions/masks to see if I could make it work, but I had no success.

## Checklist

- [X] I have checked that there is no similar issue in the repo
- [X] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/)
- [X] I have provided a minimal working example to reproduce the bug
